Support DiscoScience via new API

danielbrauer opened the pull request:
Hey, my mod Disco Science now relies on other mods to declare added ingredient colours and labs (if those labs look like the base lab). The API is explained in this FAQ:

https://mods.factorio.com/mod/DiscoScience/faq

I haven't looked through all the logic of SCT, but does this look right?

danielbrauer committed:
Support DiscoScience via new API

danielbrauer committed:
Dependency

mexmer committed:
Merge branch 'master' into discoscience

mexmer commented:
i will need to test this on current version, after that i will either merge it or close it.

KiwiHawk committed:
Add StyLuaFormatter

KiwiHawk committed:
Format Code

KiwiHawk committed:
Updating for Factorio 2.0 #98 #99

KiwiHawk committed:
Format Code

KiwiHawk committed:
Use pairs instead of ipairs

KiwiHawk committed:
Updating Bob's mods compatibility

KiwiHawk committed:
Format Code

KiwiHawk committed:
Fixing recipe name typo

KiwiHawk committed:
Merge branch 'dev' of https://github.com/mexmer/ScienceCostTweakerM into dev

KiwiHawk committed:
Remove lab scaling setting

KiwiHawk committed:
Fix prerequisites / progression

Make yellow lab / science always require purple science
Make yellow lab require underground belts to make this make sense

KiwiHawk committed:
Check if BobTech mod is enabled before overwriting setting

KiwiHawk committed:
Add explosives to military science

KiwiHawk committed:
Fix number overlay size

KiwiHawk committed:
Fixing Angel's mods compatibility

KiwiHawk committed:
Format Code

KiwiHawk committed:
High-Tech science renamed to Utility science

KiwiHawk committed:
Merge branch 'dev' of https://github.com/mexmer/ScienceCostTweakerM into dev

KiwiHawk committed:
Remove references to bio industries

Mod has not been updated for 2.0 and the fork doesn't seem to add this recipe anyway

KiwiHawk committed:
Basic omni compatibility

Only enough to make it load. I'm not spending time testing this

KiwiHawk committed:
Format Code

KiwiHawk committed:
Fix AAI Industry compatibility

KiwiHawk committed:
Update info.json

KiwiHawk committed:
Merge branch 'dev' into pr/39

KiwiHawk committed:
Fix merge

KiwiHawk committed:
Updating changelog

KiwiHawk committed:
Format Code

KiwiHawk commented:
I can't change this PR to target dev branch so I have manually merged it. Thank you! And sorry for the delay.
